### PR TITLE
Burstable QoS cgroup has cpu shares assigned

### PR DESCRIPTION
**Pull request by derekwaynecarr**
**What this PR does / why we need it**:
This PR sets the Burstable QoS cgroup cpu shares value to the sum of the pods cpu requests in that tier.  We need it for proper evaluation of CPU shares in the new QoS hierarchy.

**Special notes for your reviewer**:
It builds against the framework proposed for https://github.com/kubernetes/kubernetes/pull/41833

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
fyi @sjenning

**Comment by derekwaynecarr** [comment]
FYI @vishh @dashpole @sjenning

File: hack/local-up-cluster.sh
Comment (sjenning) [inline hack/local-up-cluster.sh]: are you meaning to change these default? or did you not mean to commit this?

File: pkg/kubelet/cm/qos_container_manager_linux.go
Comment (sjenning) [inline pkg/kubelet/cm/qos_container_manager_linux.go]: nit: `for _, pod := range pods`? then you don't need the assignment on below

File: pkg/kubelet/cm/qos_container_manager_linux.go
Comment (sjenning) [inline pkg/kubelet/cm/qos_container_manager_linux.go]: The idea was that this call would be made from the switch statement below. I see that you are wanting to set `MinShares` on BE regardless of whether or not any cpu= option was provided to `--experimental-qos-reserve-limits`, which kinda works against the flow of the code here.

What would you think about having a function that set `MinShares` for BE from `Start()` since that won't ever need updating, and then have a separate function that updates the Burstable shares according to the reserve percentage the the users specifies, which currently isn't being used to calculate the Burstable share right now (why?).

**Review by sjenning — CHANGES_REQUESTED** [review]

File: hack/local-up-cluster.sh
Comment (derekwaynecarr) [inline hack/local-up-cluster.sh]: didnt mean to commit, will remove when not WIP

File: pkg/kubelet/cm/qos_container_manager_linux.go
Comment (derekwaynecarr) [inline pkg/kubelet/cm/qos_container_manager_linux.go]: @sjenning -- we need to run a reconciliation loop to ensure that BestEffort is always 2 shares in case anyone else ever mutates it on the host w/o kubelet knowledge.

**Comment by vishh** [comment]
@derekwaynecarr I'm swamped with v1.6 PRs atm. Based on the roadmap the following PRs are still pending:
1. Node Allocatable proposal and implementation
2. Guaranteed scheduling for Critical pods
3. Pod Cgroups rollout. I'm yet to identify key metrics to manage rollout.
4. Support for multiple GPUs. 
5. Eviction improvements - pod lifecycle changes and avoiding deleting multiple pods.

I don't know if I will be able to review the QoS related PRs before code freeze. I talked to @dchen1107 and she felt overwhelmed too. One of us definitely wish to review this. 

Is QoS level enforcement critical/urgent for you and @sjenning? 
Let's sync up on the priority to avoid confusion after the code freeze.

**Comment by derekwaynecarr** [comment]
Qos level enforcement is needed for CPU shares to actually get enforced properly for Burstable tier relative to Guaranteed pods.   We can synch tomorrow.

File: pkg/kubelet/cm/qos_container_manager_linux.go
Comment (vishh) [inline pkg/kubelet/cm/qos_container_manager_linux.go]: Use `v1. PodRequestsAndLimits()` instead https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/resource_helpers.go#L185

**Review by vishh — COMMENTED** [review]
Just one comment. Otherwise this PR LGTM

**Comment by derekwaynecarr** [comment]
ha, i messed something up there... i will pick this up once seth's dependent pr merges.

**Comment by derekwaynecarr** [comment]
fixed up my error, i am marking this do-not-merge until seth's pr lands.

**Comment by derekwaynecarr** [comment]
@k8s-bot test this

**Comment by derekwaynecarr** [comment]
i am waiting to update this pr until all pertinent prereqs are merged.

i am tagging it lgtm and do-not-merge in the interim per:
https://github.com/kubernetes/kubernetes/pull/41753#pullrequestreview-23674238

**Comment by derekwaynecarr** [comment]
i rebased this pr, marking do not merge still while seth's pr lands.

**Comment by derekwaynecarr** [comment]
requisite pr has merged, will look to rebase shortly....

**Comment by derekwaynecarr** [comment]
@vishh -- just pushed the rebased code, retagging.

**Comment by vishh** [comment]
@derekwaynecarr ack. Now we wait for the merge queue to get to this PR.

**Pull request merged**
